sdk/tests/core/CMakeLists.txt

```diff
@@ -183,11 +187,6 @@ else()
                     )
 endif()
 
-target_link_libraries(az_core_test_lib_test
-  PUBLIC
-    az_iot_common
```

Comment:
I think az_iot_common might be able to be removed in the core samples cmakelists

Reply:
Indeed. Removed from samples CMakeLists.txt.